File: universal/src/main/java/com/github/fernthedev/packets/LevelUp.java
package com.github.fernthedev.packets;

import com.github.fernthedev.core.packets.Packet;
import com.github.fernthedev.core.packets.PacketInfo;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.io.Serializable;

@AllArgsConstructor
@Getter
@PacketInfo(name = "LEVEL_UP_PACKET")
public class LevelUp extends Packet implements Serializable {
    private final int level;
}

File: universal/src/main/java/com/github/fernthedev/packets/LevelUp.kt
package com.github.fernthedev.packets

import com.github.fernthedev.lightchat.core.packets.PacketInfo
import com.github.fernthedev.lightchat.core.packets.PacketJSON
import java.io.Serializable

@PacketInfo(name = "LEVEL_UP_PACKET")
class LevelUp(
    val level: Int = 0
) : PacketJSON(), Serializable